packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -0,0 +1,359 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import DatePicker from 'terra-date-picker';
+import TimeInput from 'terra-time-input';
+import DateUtil from 'terra-date-picker/lib/DateUtil';
+import styles from './DateTimePicker.scss';
+import DateTimeUtil from './DateTimeUtil';
+import TimeClarification from './_TimeClarification';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * An array of ISO 8601 string representation of the dates to disable in the picker.
+   */
+  excludeDates: PropTypes.arrayOf(PropTypes.string),
+  /**
+   * A function that gets called for each date in the picker to evaluate which date should be disabled. A return value of true will be enabled and false will be disabled.
+   */
+  filterDate: PropTypes.func,
+  /**
+   * An array of ISO 8601 string representation of the dates to enable in the picker. All Other dates will be disabled.
+   */
+  includeDates: PropTypes.arrayOf(PropTypes.string),
+  /**
+   * Custom input attributes to apply to the date and time inputs. Use the name prop to set the name for the input. Do not set the name in inputAttribute as it will be ignored.
+   */
+  inputAttributes: PropTypes.object,
+  /**
+   * An ISO 8601 string representation of the maximum date time.
+   */
+  maxDateTime: PropTypes.string,
+  /**
+   * An ISO 8601 string representation of the minimum date time.
+   */
+  minDateTime: PropTypes.string,
+  /**
+   * Name of the date input. The name should be unique.
+   */
+  name: PropTypes.string.isRequired,
+  /**
+   * A callback function to execute when a valid date is selected or entered. The first parameter is the event. The second parameter is the changed date time value.
```

Comment:
Is the onChange only triggered with a valid date selection or a valid date and time combination?

Reply:
onChange is triggered when there is a valid date without a time or a valid date with a valid time. If no time is entered, the default time passed to onChange is 00:00. It does not get triggered with only a valid time and no date.